pandas/tests/reshape/test_crosstab.py

```diff
@@ -534,16 +534,50 @@ def test_crosstab_with_numpy_size(self):
         tm.assert_frame_equal(result, expected)
 
     def test_crosstab_dup_index_names(self):
-        # GH 13279
-        s = Series(range(3), name="foo")
+        # We test that duplicated column names do not produce issues
+        # GH Issue: #22529, GH#13279
 
-        result = crosstab(s, s)
-        expected_index = Index(range(3), name="foo")
-        expected = DataFrame(
-            np.eye(3, dtype=np.int64), index=expected_index, columns=expected_index
+        # Duplicate name shared between rows and columns
+        s1 = Series(range(3), name="foo")
+        s2 = s1 + 1
+        expected = crosstab(s1, s2.rename("bar")).rename_axis(
+            columns={"bar": "foo"}, axis=1
+        )
+        result = crosstab(s1, s2)
+        tm.assert_frame_equal(result, expected)
+        assert result.index.names == ["foo"]
+        assert result.columns.names == ["foo"]
+
+        # Row names duplicated
+        s1 = Series(range(3), name="foo")
+        s2 = s1 + 1
+        s3 = Series(range(3), name="bar_col")
+
+        expected = crosstab([s1, s2.rename("bar")], s3).rename_axis(
+            index={"bar": "foo"}, axis=0
+        )
+        result = crosstab([s1, s2], s3)
+
+        tm.assert_frame_equal(result, expected)
+
+        assert result.index.names == ["foo", "foo"]
+        assert result.columns.names == ["bar_col"]
+
+        # Column names duplicated
```

Comment:
can you split this into a separate test (and rename both appropriately)

Reply:
Do you want duplicated column names as a separate test? In that case, I think it makes more sense to do three tests:
- One for when the name is shared between rows and columns
- Duplicated row names
- Duplicated column names

Makes sense?

Reply:
Renamed the test. Let me know if you prefer it split into 3.

Reply:
yes prefer this to be split